components/brave_new_tab_ui/helpers/newTabUtils.ts

```diff
@@ -8,3 +8,18 @@ export const isHttpOrHttps = (url?: string) => {
   }
   return /^https?:/i.test(url)
 }
+
+/**
+ * Obtains a letter / char that represents the current site
+ * @param site - The site requested from the top site's list
+ */
+export const getLetterFromSite = (site: NewTab.Site) => {
+  let name
+  try {
+    name = new window.URL(site.url || '').hostname
```

Comment:
since the function will ignore this line if `site.title` exists, can we only perform this object construction if `!site.title`?